demos/text_detection_demo/src/text_recognition.cpp

```diff
@@ -25,6 +26,27 @@ namespace  {
         }
         *prob = 1.0f / static_cast<float>(sum);
     }
+
+    std::vector<float> softmax(const std::vector<float>::const_iterator& begin, const std::vector<float>::const_iterator& end) {
+        std::vector<float> prob(end - begin, 0.f);
+        std::transform(begin, end, prob.begin(), static_cast<double(*)(double)>(std::exp));
```

Comment:
```suggestion
        std::transform(begin, end, prob.begin(), [](float x) { return std::exp(x); });
```
The current draft of the C++ standard [prohibits](http://eel.is/c++draft/namespace.std#6) forming a pointer to most standard library functions. It would be best to start avoiding doing that now. The lambda is shorter anyway.

I've also changed `double` to `float`, since `prob` is a `vector<float>`.